x-pack/platform/plugins/shared/stack_connectors/public/connector_types/inference/connector.tsx

```diff
@@ -5,384 +5,26 @@
-      isProviderPopoverOpen,
-    ]
-  );
-
-  return (
-    <>
-      <UseField
-        path="config.provider"
-        config={{
-          validations: [
-            {
-              validator: fieldValidators.emptyField(i18n.PROVIDER_REQUIRED),
-              isBlocking: true,
-            },
-          ],
-        }}
-      >
-        {(field) => {
-          const { isInvalid, errorMessage } = getFieldValidityAndErrorMessage(field);
-          const selectInput = providerSuperSelect(isInvalid);
-          return (
-            <EuiFormRow
-              id="providerSelectBox"
-              fullWidth
-              label={
-                <FormattedMessage
-                  id="xpack.stackConnectors.components.inference.providerLabel"
-                  defaultMessage="Service"
-                />
-              }
-              isInvalid={isInvalid}
-              error={errorMessage}
-            >
-              <EuiInputPopover
-                id={'popoverId'}
-                fullWidth
-                input={selectInput}
-                isOpen={isProviderPopoverOpen}
-                closePopover={handleProviderClosePopover}
-                className="rightArrowIcon"
-              >
-                <SelectableProvider
-                  isLoading={isLoading}
-                  getSelectableOptions={getProviderOptions}
-                  onClosePopover={handleProviderClosePopover}
-                  onProviderChange={onProviderChange}
-                />
-              </EuiInputPopover>
-            </EuiFormRow>
-          );
-        }}
-      </UseField>
-      {config?.provider ? (
-        <>
-          <EuiSpacer size="m" />
-          <ConnectorConfigurationFormItems
-            itemsGrow={false}
-            isLoading={false}
-            direction="column"
-            items={requiredProviderFormFields}
-            setConfigEntry={onSetProviderConfigEntry}
-          />
-          <EuiSpacer size="m" />
-          <AdditionalOptionsConnectorFields
-            config={config}
-            readOnly={readOnly}
-            isEdit={isEdit}
-            optionalProviderFormFields={optionalProviderFormFields}
-            onSetProviderConfigEntry={onSetProviderConfigEntry}
-            onTaskTypeOptionsSelect={onTaskTypeOptionsSelect}
-            taskTypeFormFields={taskTypeFormFields}
-            taskTypeOptions={taskTypeOptions}
-            selectedTaskType={selectedTaskType}
-          />
-          <EuiSpacer size="l" />
-          <EuiHorizontalRule />
-          {getProviderSecretsHiddenField(
-            providerSchema,
-            setRequiredProviderFormFields,
-            isSubmitting
-          )}
-          {getProviderConfigHiddenField(
-            providerSchema,
-            setRequiredProviderFormFields,
-            isSubmitting
-          )}
-        </>
-      ) : null}
-    </>
-  );
+  return !isLoading ? (
+    <InferenceServiceFormFields providers={providers ?? []} isEdit={isEdit} />
```

Comment:
since you're using the shared components now and it looks like the shared components have their own routes for getting the services configuration (hard-coded for now), should we also remove the server side routes in the stack_connectors folder?